src/Simplify_Add.cpp

```diff
@@ -49,14 +49,19 @@ Expr Simplify::visit(const Add *op, ExprInfo *bounds) {
         // clang-format off
         if (EVAL_IN_LAMBDA
             (rewrite(x + x, x * 2) ||
-             rewrite(ramp(x, y) + ramp(z, w), ramp(x + z, y + w), is_same_type(x, z)) ||
-             rewrite(ramp(x, y) + broadcast(z), ramp(x + z, y), is_same_type(x, z)) ||
-             rewrite(broadcast(x) + broadcast(y), broadcast(x + y), is_same_type(x, y)) ||
-             rewrite((x + broadcast(y)) + broadcast(z), x + broadcast(y + z), is_same_type(y, z)) ||
-             rewrite((x - broadcast(y)) + broadcast(z), x + broadcast(z - y), is_same_type(y, z)) ||
+             rewrite(ramp(x, y, c0) + ramp(z, w, c0), ramp(x + z, y + w, c0)) ||
```

Comment:
@jn80842 @shoaibkamil Simplifier rules with nested vectors are starting to get non-trivial

Reply:
Yeah, we should add support for verifying these.